src/thunderbots/software/ai/navigator/obstacle.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "geom/util.h"
+#include "ai/world/world.h"
+
+constexpr double DEFAULT_AVOID_DIST = 0.15;  // meters
+
+class RobotObstacle
+{
+   public:
+    /**
+     * Constructor.
+     *
+     * @param r Robot to create obstacle from.
+     */
+    RobotObstacle(Robot& r, double avoid_dist);
+
+    /**
+     * Returns the violation distance of the given point inside the obstacle
+     * boundary; 0.0 if no violation.
+     *
+     * @param p point to check if violating
+     *
+     * @return Violation distance, defined as distance from the nearest boundary
+     *         to the given point; 0.0 if no violation occured.
+     */
+    double getViolation(Point& p);
+
+    /**
+     * Returns the closest Point on the obstacle boundary to the given Point that
+     * is not inside the boundary. If the given Point does not violate the boundary,
+     * the given Point will be returned.
+     *
+     * @param p Point to check for
+     * @return A point on the boundary nearest to p if p violates the boundary, otherwise
+     *         returns p
+     */
+    Point getNearestValidPoint(Point& p);
+
+    /**
+     * TODO: This function may have to be rewritten depending on our needs later.
+     * Checks if the given Robot will collide with the obstacle based on the current
+     * velocities.
+     *
+     * @param r Robot to check against.
+     * @return true if collision possible, false otherwise.
+     */
+    bool willCollide(Robot& r);
```

Comment:
This is probably a discussion that we need to have over the whole codebase, but I'd advocate for almost always being verbose in naming (in this case `r` -> `robot`). Using single letters like this obfuscates things, reduces readability, and doesn't really save much space. 

Same comment goes for other variables here (such as `p` -> `point`).

@MathewMacDougall thoughts? Either way we should standardize how we write variable names.

Reply:
Mix feelings on this. Maybe use the longer names when the variable is a function parameter or something similar with a potentially very large scope (so you don't have to scroll to double check what it is). In loops I'm more ok with single letter variable names as long as the loops aren't very large, since the scope is relatively small. When in doubt, be more verbose. It's easier to add more detail now than later.

Reply:
Ok sounds good, I'll use the long form parameter name.